### PR TITLE
Adding Zicbom extension support

### PR DESCRIPTION
**Pull request by jleidel**
Adding Zicbom extension support (PR #225)
Adds:

- cmo.flush
- cmo.inval
- cmo.clean

### CODE REVIEW

File: include/insns/Zicbom.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// All Rights Reserved
+// contact@tactcomplabs.com
+//
+// See LICENSE in the top level directory for licensing details
+//
+
+#ifndef _SST_REVCPU_ZICBOM_H_
+#define _SST_REVCPU_ZICBOM_H_
+
+#include "../RevInstHelpers.h"
```
Comment (rkabrick) [inline include/insns/Zicbom.h:14]: Do you need the relative header locations?

File: include/insns/Zicbom.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+#include <vector>
+#include <limits>
+
+namespace SST::RevCPU{
+class Zicbom : public RevExt{
+
+  static bool cmo(RevFeature *F, RevRegFile *R, RevMem *M, RevInst Inst) {
+    switch(Inst.imm){
+    case 0b000000000000:
```
Comment (rkabrick) [inline include/insns/Zicbom.h:25]: I know you commented the case names but maybe put them in either macros or an enum instead?

Comment (rkabrick) [inline include/insns/Zicbom.h:25]: Same enum question

File: test/zicbom/rev-test-zicbom.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (C) 2017-2023 Tactical Computing Laboratories, LLC
+# All Rights Reserved
+# contact@tactcomplabs.com
+#
+# See LICENSE in the top level directory for licensing details
+#
+# rev-test-cache2.py
```
Comment (rkabrick) [inline test/zicbom/rev-test-zicbom.py:8]: Doesn't look like a cache test to me 🤔

File: test/zicbom/zicbom.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
+/*
+ * ex2.c
+ *
+ * RISC-V ISA: RV64I
```
Comment (rkabrick) [inline test/zicbom/zicbom.c:4]: RV64GC_Zicbom & not ex2.c

**Review by rkabrick — CHANGES_REQUESTED** [review]
Everything logically looks good... just look at the minor comments and do with them as you see fit

File: include/insns/Zicbom.h (hunk 1 above)
Comment (jleidel) [reply include/insns/Zicbom.h:14]: matches the other insn implementation headers

File: include/insns/Zicbom.h (hunk 2 above)
Comment (jleidel) [reply include/insns/Zicbom.h:25]: fixed in d8b98be

Comment (jleidel) [reply include/insns/Zicbom.h:25]: fixed in d8b98be

File: test/zicbom/rev-test-zicbom.py (hunk 3 above)
Comment (jleidel) [reply test/zicbom/rev-test-zicbom.py:8]: fixed in d8b98be

File: test/zicbom/zicbom.c (hunk 4 above)
Comment (jleidel) [reply test/zicbom/zicbom.c:4]: fixed in d8b98be

**Review by rkabrick — APPROVED** [review]
LGTM

**Comment by jleidel** [comment]
code updated in d8b98be; CI tests running now

**Pull request merged**